builtin/audit/socket/backend.go

```diff
@@ -0,0 +1,116 @@
+package socket
+
+import (
+	"bytes"
+	"fmt"
+	"net"
+	"strconv"
+
+	"github.com/hashicorp/vault/audit"
+	"github.com/hashicorp/vault/logical"
+)
+
+func Factory(conf *audit.BackendConfig) (audit.Backend, error) {
+	if conf.Salt == nil {
+		return nil, fmt.Errorf("Nil salt passed in")
+	}
+
+	address, ok := conf.Config["address"]
```

Comment:
I think I'd actually prefer if we error out here if no address is passed in, because I can't think of a particularly sensible default.